server/src/api/v2/event-registrations/register-for-event.ts

```diff
@@ -4,17 +4,39 @@ import { mapEventRegistration } from '../../../functions/map-event-registration.
 import { wrapResponse } from '../../../functions/response-wrapper';
 import { EventRegistrationDataSnapshot, RawEventRegistrationData } from '../../../interfaces/event-registration.interface';
 import { EventRegistration } from '../../../models/event-registration.model';
+import { Event } from '../../../models/event.model';
+import { Vars } from '../../../vars';
 
 export async function registerForEvent(req: Request, res: Response): Promise<Response> {
     let success = true;
     const incomingData: EventRegistrationDataSnapshot = req.body;
     const mappedIncomingData: RawEventRegistrationData = mapEventRegistration(incomingData, req.params.id);
-    
+
     const requiredFields = EventRegistration.requiredFields();
     if (!objectHasRequiredAndNotEmptyKeys(mappedIncomingData, requiredFields)) {
         return res.status(400).send(wrapResponse(false, { error: 'Not all required fields have been set' }));
     }
 
+    const event: null | { rows: Event[], count: number } = await Event.scope(['full', { method: ['onlyCurrentOrg', Vars.currentOrganization.id] }]).findAndCountAll(
+        {
+            where: {
+                id: mappedIncomingData.event_id
+            }
+        })
+        .catch(() => {
+            success = false;
+            return null;
+        });
+    if (!success) {
+        return res.status(500).send(wrapResponse(false, { error: 'Database error' }));
+    }
+    if (event === null) {
+        return res.status(400).send(wrapResponse(false, { error: 'No Event with given id found!' }));
+    }
+    if (event.rows[0].max_participants !== null && event.count >= event.rows[0].max_participants){
+        return res.status(400).send(wrapResponse(false, { error: 'The participation limit has been reached!' }));
```

Comment:
Können wir das auch in der normalen API ausgeben? Max_people_count_reached: true beim get event

Reply:
Besser wäre ein Count der belegten Plätze